test/NuGet.Core.Tests/NuGet.Protocol.Tests/Plugins/PluginDiscovererTests.cs

```diff
@@ -153,5 +176,544 @@ public async Task DiscoverAsync_IsIdempotent()
+            File.Create(nonPluginFilePath);
+            var fileInfo = new FileInfo(nonPluginFilePath);
+
+            // Act
+            bool result = PluginDiscoverer.IsValidPluginFile(fileInfo);
+
+            // Assert
+            Assert.False(result);
+        }
+
+        [PlatformFact(Platform.Linux)]
+        public void IsValidPluginFile_Unix_ExecutableFile_ReturnsTrue()
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+            var workingPath = testDirectory.Path;
+            var pluginFilePath = Path.Combine(workingPath, "plugin");
+            File.Create(pluginFilePath).Dispose();
+
+#if NET8_0_OR_GREATER
+            // Set execute permissions
+            File.SetUnixFileMode(pluginFilePath, UnixFileMode.UserExecute | UnixFileMode.UserRead);
+#else
+            // Use chmod to set execute permissions
+            var process = new Process();
+            process.StartInfo.FileName = "/bin/bash";
+            process.StartInfo.Arguments = $"-c \"chmod +x {pluginFilePath}\"";
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.RedirectStandardOutput = true;
+            process.Start();
+            process.WaitForExit();
+#endif
+
+            var fileInfo = new FileInfo(pluginFilePath);
+
+            // Act
+            bool result = PluginDiscoverer.IsValidPluginFile(fileInfo);
+
+            // Assert
+            Assert.True(result);
+        }
+
+#if !NET8_0_OR_GREATER
+        [PlatformFact(Platform.Linux)]
+        public void IsExecutable_FileIsExecutable_ReturnsTrue()
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+            var workingPath = testDirectory.Path;
+            var pluginFilePath = Path.Combine(workingPath, "plugin");
+            File.Create(pluginFilePath);
+
+            // Set execute permissions
+            var process = new Process();
+            process.StartInfo.FileName = "/bin/bash";
+            process.StartInfo.Arguments = $"-c \"chmod +x {pluginFilePath}\"";
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.RedirectStandardOutput = true;
+            process.Start();
+            process.WaitForExit();
+
+            var fileInfo = new FileInfo(pluginFilePath);
+
+            // Act
+            bool result = PluginDiscoverer.IsExecutable(fileInfo);
+
+            // Assert
+            Assert.True(result);
+        }
+
+        [PlatformFact(Platform.Linux)]
+        public void IsExecutable_FileIsNotExecutable_ReturnsFalse()
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+            var workingPath = testDirectory.Path;
+            var pluginFilePath = Path.Combine(workingPath, "plugin");
+            File.Create(pluginFilePath);
+
+            // Remove execute permissions
+            var process = new Process();
```

Comment:
There is an opportunity to refactor here because we set/remove execute permissions for files in all the new tests. I think we can create a private utility method and invoke it from the tests as needed.